redis-om-spring/src/main/java/com/redis/om/spring/RedisModulesConfiguration.java

```diff
@@ -284,43 +285,51 @@ private List<Field> findIndexFields(java.lang.reflect.Field field, String prefix
-      if (CharSequence.class.isAssignableFrom(field.getType()) || (field.getType() == Boolean.class)) {
+      if (CharSequence.class.isAssignableFrom(fieldType) || (fieldType == Boolean.class)) {
         fields.add(indexAsTagFieldFor(field, isDocument, prefix, indexed.sortable(), indexed.separator(),
             indexed.arrayIndex()));
       }
       //
       // Any Numeric class -> Numeric Search Field
       //
-      else if (Number.class.isAssignableFrom(field.getType()) || (field.getType() == LocalDateTime.class)
+      else if (Number.class.isAssignableFrom(fieldType) || (fieldType == LocalDateTime.class)
           || (field.getType() == LocalDate.class) || (field.getType() == Date.class)) {
         fields.add(indexAsNumericFieldFor(field, isDocument, prefix, indexed.sortable(), indexed.noindex()));
       }
       //
       // Set / List
       //
-      else if (Set.class.isAssignableFrom(field.getType()) || List.class.isAssignableFrom(field.getType())) {
+      else if (Set.class.isAssignableFrom(fieldType) || List.class.isAssignableFrom(fieldType)) {
         Optional<Class<?>> maybeCollectionType = com.redis.om.spring.util.ObjectUtils.getCollectionElementType(field);
 
         // It is only possible to index an array of strings or booleans in a TAG identifier.
         // https://redis.io/docs/stack/json/indexing_json/#json-arrays-can-only-be-indexed-in-tag-identifiers
-        if (maybeCollectionType.isPresent() && (CharSequence.class.isAssignableFrom(maybeCollectionType.get()) || (maybeCollectionType.get() == Boolean.class))) {
-          fields.add(indexAsTagFieldFor(field, isDocument, prefix, indexed.sortable(), indexed.separator(),
+        if (maybeCollectionType.isPresent()) {
+          Class<?> collectionType = maybeCollectionType.get();
+          
+          if (CharSequence.class.isAssignableFrom(collectionType) || (collectionType == Boolean.class)) {
+            fields.add(indexAsTagFieldFor(field, isDocument, prefix, indexed.sortable(), indexed.separator(),
               indexed.arrayIndex()));
-        // Index nested fields
-        } else if (isDocument) {
-          // Index nested JSON Array field. But the current implementation of RediSearch supports array only for TAG fields, not TEXT fields.
-          // https://github.com/RediSearch/RediSearch/issues/2293
-          logger.info(String.format("FOUND @Nested annotation on field of type: %s", field.getType()));
-          fields.addAll(indexAsNestedFieldFor(field, prefix));
+          // Index nested fields
+          } else if (isDocument) {
+            // Index nested JSON Array field. But the current implementation of RediSearch supports array only for TAG fields, not TEXT fields.
+            // https://github.com/RediSearch/RediSearch/issues/2293
+            logger.debug(String.format("FOUND nested field on field of type: %s", field.getType()));
+            fields.addAll(indexAsNestedFieldFor(field, prefix));
+          }
+        } else {
```

Comment:
this else seems unnecessary, unless you want to log an error or throw an exception because the option didn't unwind with a type.